### PR TITLE
Commander: add options for max-wind and max-flight-time RTL

### PR DESCRIPTION
**Pull request by sfuhrer**
**Describe problem solved by this pull request**
- no option atm for flight time restriction (it's a legal requirement in some countries to restrict the flight time of a drone)
- no option to abort the flight if the current wind speeds (estimated on board) are above the vehicle thresholds. 

**Describe your solution**
Add the above two features to Commander. Combine the Wind abort with the wind warning that already existed. 
Further I removed the high_wind state from the failure detector. At some point it was planned to add the high wind abort through it, but as it actually isn't a system failure it should be separated. 

**Test data / coverage**
SITL tested.

### CODE REVIEW

File: src/modules/commander/commander_params.c [hunk 1]
```diff
@@ -1065,3 +1065,38 @@ PARAM_DEFINE_INT32(COM_ARM_SDCARD, 1);
...
+ *
+ *
+ * @unit s
+ * @min 0
+ * @max 10000
+ * @value 0 Disable
+ * @group Commander
+ * @unit s
+ */
+PARAM_DEFINE_INT32(COM_FLT_TIME_MAX, 3540);
```
[the remark this answers is not in the document]
Comment (sfuhrer) [reply src/modules/commander/commander_params.c:1085]: Yes I though about that as well after making the PR, will disable it by default.

File: src/modules/commander/Commander.cpp [hunk 2]
```diff
@@ -2906,13 +2906,32 @@ Commander::run()
...
 
 		_status_flags.flight_terminated = _armed.force_failsafe || _armed.lockdown || _armed.manual_lockdown;
 
+		// Trigger RTL if flight time is larger than max flight time specified in COM_FLT_TIME_MAX.
+		// The user is not able to override once above threshold, excpet for triggering Land.
+		if (!_vehicle_land_detected.landed
+		    && _param_com_flt_time_max.get() > 0
+		    && _internal_state.main_state != commander_state_s::MAIN_STATE_AUTO_RTL
+		    && _internal_state.main_state != commander_state_s::MAIN_STATE_AUTO_LAND
+		    && hrt_elapsed_time(&_status.takeoff_time) > (1_s * _param_com_flt_time_max.get())) {
```
[the remark this answers is not in the document]
Comment (sfuhrer) [reply src/modules/commander/Commander.cpp:2923]: done

File: src/modules/commander/commander_params.c [hunk 3]
```diff
@@ -1065,3 +1065,38 @@ PARAM_DEFINE_INT32(COM_ARM_SDCARD, 1);
...
  */
 PARAM_DEFINE_FLOAT(COM_WIND_WARN, -1.f);
+
+/**
+ * Maximum allowed flight time
+ *
+ * The vehicle returns to launch when the vehicle is not landed and
+ * the time since takeoff is above this value.
+ *
+ * The default is 3540 seconds (59 minutes). Set a value of zero to disable.
```
[the remark this answers is not in the document]
Comment (sfuhrer) [reply src/modules/commander/commander_params.c:1075]: done

File: msg/failure_detector_status.msg [hunk 4]
```diff
@@ -6,7 +6,6 @@ bool fd_pitch
 bool fd_alt
 bool fd_ext
 bool fd_arm_escs
-bool fd_high_wind
```
[the remark this answers is not in the document]
Comment (sfuhrer) [reply msg/failure_detector_status.msg:9]: done

File: src/modules/commander/Commander.cpp [hunk 5]
```diff
@@ -4469,13 +4487,30 @@ void Commander::checkWindAndWarn()
...
+		    && _internal_state.main_state != commander_state_s::MAIN_STATE_AUTO_RTL
+		    && _internal_state.main_state != commander_state_s::MAIN_STATE_AUTO_LAND) {
 			mavlink_log_critical(&_mavlink_log_pub, "High wind speed detected (%.1f m/s), landing advised\t", (double)wind.norm());
 
 			events::send<float>(events::ID("commander_high_wind_warning"),
 			{events::Log::Warning, events::LogInternal::Info},
 			"High wind speed detected ({1:.1m/s}), landing advised", wind.norm());
 			_last_wind_warning = hrt_absolute_time();
+
+		} else if (_param_com_wind_max_rtl.get() > FLT_EPSILON
```
[the remark this answers is not in the document]
Comment (sfuhrer) [reply src/modules/commander/Commander.cpp:4502]: make sense

[the remark this answers is not in the document]
Comment (sfuhrer) [reply src/modules/commander/Commander.cpp]: done

File: src/modules/commander/commander_params.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (sfuhrer) [reply src/modules/commander/commander_params.c]: done

File: src/modules/commander/commander_params.c [hunk 6]
```diff
@@ -1065,3 +1065,38 @@ PARAM_DEFINE_INT32(COM_ARM_SDCARD, 1);
...
+ * A negative value disables the feature.
+ *
+ * @min -1
+ * @max 30
+ * @decimal 1
+ * @increment 0.1
+ * @group Commander
+ * @unit m/s
+ */
+PARAM_DEFINE_FLOAT(COM_WIND_MAX_RTL, -1.f);
```
[the remark this answers is not in the document]
Comment (sfuhrer) [reply src/modules/commander/commander_params.c]: done

**Comment by sfuhrer** [comment]
@MaEtUgR thanks for the review, I agree with you on all points :) Addressed in https://github.com/PX4/PX4-Autopilot/pull/19578/commits/717521f0b73042d12b48d987a472c320a9eee3e2

File: src/modules/commander/Commander.cpp [hunk 7]
```diff
@@ -4469,13 +4487,33 @@ void Commander::checkWindAndWarn()
...
+			   && _internal_state.main_state != commander_state_s::MAIN_STATE_AUTO_RTL
+			   && _internal_state.main_state != commander_state_s::MAIN_STATE_AUTO_LAND) {
+
 			mavlink_log_critical(&_mavlink_log_pub, "High wind speed detected (%.1f m/s), landing advised\t", (double)wind.norm());
 
 			events::send<float>(events::ID("commander_high_wind_warning"),
 			{events::Log::Warning, events::LogInternal::Info},
 			"High wind speed detected ({1:.1m/s}), landing advised", wind.norm());
 			_last_wind_warning = hrt_absolute_time();
+
```
Comment (MaEtUgR) [inline src/modules/commander/Commander.cpp:4516]:
```suggestion
```

**Review by MaEtUgR — APPROVED** [review]
Thanks for addressing the comments 👍

File: src/modules/commander/Commander.cpp (hunk 7 above)
Comment (sfuhrer) [reply src/modules/commander/Commander.cpp]: addressed and force pushed

**Review by MaEtUgR — APPROVED** [review]

**Pull request merged**